Person Content Type MVP

akslay opened the pull request:
- Closes #116
- Closes #117
- Closes #380 

This pull requests includes the content type, views, view modes, templates, config, etc. necessary to replicate the Person content type from Drupal 7.

Probo build: https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-398.probo.build/ (updated)

akslay commented:
**Issues & topics related to decisions on this PR:**

- Missing contextual link to reorder content (https://www.drupal.org/project/draggableviews/issues/2843858), so using a code-based block as interim solution since it can be easily removed down the road.
- Relationship and/or distinct not working properly and causing multiple duplicates (https://www.drupal.org/project/drupal/issues/3037255, https://www.drupal.org/project/entity_reference_revisions/issues/2799479, https://www.drupal.org/project/drupal/issues/2993688), so I have just left the relationship out for now, since it seems to work without it.
- Can't expose title field for view modes (https://www.drupal.org/project/contrib_tracker/issues/2607758)
- Overlapping/unusable contextual links (https://www.drupal.org/project/drupal/issues/2983655), so I have an interim solution which utilizes CSS to hide the media contextual link.
- Applied a patch to fix missing "attach" option to set context for draggableviews weight (https://www.drupal.org/files/issues/2019-12-17/draggableviews-sort_handler_specify_order_view-2767437-66.patch)

**Some potential follow-up items:**
- Currently no option included to crop images--wasn't sure which direction we planed to go for this module-wise since other aspects in AZQS will be using crop.
- Need to revisit grouping aspect for reorder/sort view, as grouping wasn't working as intended for nodes with more than 1 category selected

akslay commented:
Looks like the card images aren't at the same height. Might be because we're not using the `.card-group` class and are using the Bootstrap grid/columns instead.

danahertzberg commented:
I recommend not closing the epic (#298) with this PR in case we need to add any follow up tickets.

joeparsons reviewed (commented):
We'll definitely need to remove the drupal8.xxxxx.sql.gz file that are currently being added here before merging.

akslay commented:
**Follow-up issues:**
- [ ] Add in contextual filter for term category
- [ ] Re-visit grouping in the reorder/sort view
- [ ] Add Chosen to install profile
- [ ] Add cropping module to install profile

akslay commented:
> We'll definitely need to remove the drupal8.xxxxx.sql.gz file that are currently being added here before merging.

Thanks for catching those--forgot they were in there from development.

danahertzberg commented:
Here are my notes/questions.

**General**
1. Are the Directory Grid and Directory List menu items a result of the view or were they added manually to the probo site?

**Person Content Type**
1. The dropdown title should be `Categories` not `Category`
2. I recommend making the Categories section collapsed by default. Then when we get the Chosen module (or similar) installed, we can reassess making it expanded by default.
3. Can the Link Text field come before the URL field?
4. I'm not seeing the Summary fields section to define the Link to person and URL
5. How did you develop the display modes? I'm not seeing them under Manage Displays.

**AZ Person view - Grid Page**
1. Bootstrap grid settings don't match QS1 (columns per row and added row class)
2. No contextual filters (should also have follow up ticket for discussing how to best implement contextual filters)

**AZ Person view - Row Page**
1. Unformatted list settings don't match QS1 (Remove tags from rendered output and Add views to rows classes)
2. No contextual filters

**AZQS Reorder view - Reorder People**
1. Is it best practice to have a general reorder view or specific/separate reorder views based on content type to match QS1?
2. Should be titled "AZ Reorder"
3. Permission should be set to be role based - admin & content admin only
4. Doesn't meet ticket requirements (#380)

danahertzberg reviewed (changes requested):
See previous comment. Basic requirements are not met for all 3 tickets this PR aims to resolve.

smayden commented:
This looks great! I was able to add both the row and grid to a Page using the Add View paragraph.

Should the block to reorder people eventually be added to /people-list in addition to /people?

camikazegreen commented:
@danahertzberg It looks like the pages are created by the view, but the menu items were added to Probo for the purposes of Review. 

We have follow up tickets for Contextual Filters and improving the Reorder View (although I think that it's valuable to capture the work already completed for the Reorder View). 

All the rest of your review comments seem pretty straightforward and make sense to get resolved as part of this PR.

danahertzberg commented:
> @danahertzberg It looks like the pages are created by the view, but the menu items were added to Probo for the purposes of Review.
> 
> We have follow up tickets for Contextual Filters and improving the Reorder View (although I think that it's valuable to capture the work already completed for the Reorder View).
> 
> All the rest of your review comments seem pretty straightforward and make sense to get resolved as part of this PR.

@camikazegreen Makes sense about the pages. Just was confusing since there wasn't any person data added for review.

The contextual filters is just to match QS1. That should really be included in this PR since that's the main goal. 

Can the reorder view be removed from this PR so we can address it separately? I'm not a fan of modifying the ticket requirements just to match the PR.

akslay commented:
@danahertzberg Thanks for all your review/feedback; I've made some changes to the PR for you. However, there are a few items that remain unchanged, which are noted below along with some answers to your questions. The numbers will correspond to your own for clarity. 

**General**
1. I added those links for review purposes only for quick access to the two views. They are not part of the PR. I did have Person content added for review purposes, but the content gets reaped every time probo rebuilds, and it was rebuilt a few times since I was merging in changes added to main. 🙂

**Person Content Type**
2. I kept the categories tab open since it was a low hanging fruit for improvements to user experience. The collapsed by default categories is one of the biggest issues our editors have had, as they're unaware that the categories are located there--it's one of the most common questions we get from editors. So I left it open by default to improve that aspect of user experience since it was a quick/minor change. 

3. Unfortunately it doesn't look like we have the option to choose whether the URL or title field comes first--it's how the field type/module was built.

4. I didn't add the summary fields to this PR because that's a feature that other content types use, so I figured that would be something that's added later on as a separate feature (similar to cropping module).

5. The display modes are templates--you won't be able to modify under "Manage Display". We had a conversation during the Friday meeting that this isn't desirable (especially for the SASG team), but it was the only to get things the way we need them at this point in time. Manage Display combined with Field Groups module didn't allow for the flexibility needed in order to develop the display modes for the content type. There are some active issues to keep an eye on that I noted, that will allow us that flexibility (to a certain extent). I think during the meeting @tadean might have mentioned the possibility of adding or building something that will allow for that flexibility. (Please correct me if I'm wrong, Troy). But currently, the only way to get that flexibility would be to use something like Display Suite.

**AZ Person view - Grid Page**
1. I'm assuming this is in reference to the `.pb-4` class I added to the view? I added in the padding class to each row because things were a little too close together, especially on mobile. But I went ahead and removed that for you. 

2. I added 1 of the 2 contextual filters, but I left the second one out due to the issue with the Relationship causing duplicates that I mentioned during the PR review. There are some active issues out for this problem as well, which are noted in the PR comments. Adding that second contextual filter will probably have to be a follow-up PR once the issue with relationships/distinct filtering is corrected.

**AZ Person view - Row Page**
2. Same as AZ Grid Page above.

**AZQS Reorder view - Reorder People**
1. I thought it was better to have a single view (Reorder) with different view displays (Person, Carousel, etc.) to handle all of the ordering, as you're creating a single view and keeping everything in one place. But would be interested in getting input from @joeparsons @trackleft @tadean (etc) to see what their viewpoints on this are.

4. The requirements for this one probably won't be able to be met at this point in time, but hopefully they can be in the future. As with the current version of UAQS, you can only have one weight associated with a single node. So even if someone is in 2 categories, the node will only store 1 weight (I believe the first one). I mentioned during the PR review that the grouping functionality was not working as expected, which is why I am using the listed view of the categories as an interim solution until we have something that works better. I did revisit this issue for you (just to make sure), but there are still some pending issue(s) out that would affect the reorder view (here's one: https://www.drupal.org/project/draggableviews/issues/2867348). A quick rundown of the problem: draggableviews handles aren't showing up properly when you use grouping. It was only showing up for the first group, and every remaining group had no handle or had a row weight option instead. After some cache clears and further testing, not even the first item had a handle anymore and you only get row weights instead of handles. I've attached 2 screenshots to this comment illustrating this problem. The one labeled "issue" shows the draggable handle problem, and the one labeled "interim solution" is what I am suggesting we use in the mean time in this PR.

---------------

For your later comment, about removing the reorder view portion and adding it to a separate PR, while I do agree that it's easier to focus on smaller items at a time, I think separating it out at this point in the process would be counter productive. The individual issues were created after I had already started development on this project and had gotten pretty far along, so unfortunately things are included in a single pull request in this case. There was also a bit of a mixup on the original issue as I didn't realize it was an epic vs. a ticket (sorry, still getting used to GitHub). I have taken note of this though so that next time everything can be in separate pull requests. 🙂 Thanks again for your thorough review on this.

akslay commented:
@smayden Thanks for catching that! I have updated the PR so the block is on both pages.

joeparsons commented on themes/custom/az_barrio/az_barrio.theme:
```diff
@@ -436,3 +436,16 @@ function az_barrio_footer_logo() {
+    unset($variables['attributes']['height']);
+    $variables['attributes']['class'] = ['card-img-top'];
+  }
+  // dpm($variables);
```
It'd be good to remove this `dpm()` line before merging.

danahertzberg reviewed (approved):
Approved
